lib/bolt/cli.rb

```diff
@@ -409,15 +409,7 @@ def execute(options)
                 outputter.print_event(node, event)
               end
             when 'task'
-              task_name = options[:object]
-
-              path, metadata = load_task_data(task_name, @config[:modulepath])
-              input_method = metadata['input_method']
-
-              input_method ||= 'both'
-              executor.run_task(
-                nodes, path, input_method, options[:task_options]
-              ) do |node, event|
+              execute_task(executor, options) do |node, event|
```

Comment:
If the execute_task returns an `ExecutionResult`, the passing of a block would be unnecessary. Instead, the `ExecutionResult` could be presented to the user in some standardized way (iterating over the result of each node, etc.).

Reply:
As discussed in HipChat, the block is used to report progress, rather than return the result.